internal/pkg/manifest/workload.go

```diff
@@ -560,30 +561,45 @@ func UnmarshalWorkload(in []byte) (WorkloadManifest, error) {
 		if err := yaml.Unmarshal(in, m); err != nil {
 			return nil, fmt.Errorf("unmarshal to load balanced web service: %w", err)
 		}
+		if err := m.WindowsCompatibility(); err != nil {
+			return nil, err
+		}
 		return m, nil
 	case RequestDrivenWebServiceType:
 		m := newDefaultRequestDrivenWebService()
```

Comment:
A refactor suggestion:
```go
type manifest interface {
	WorkloadManifest
	windowsCompatibility() error
}

var m manifest
switch typeVal {
case LoadBalancedWebServiceType:
	m := newDefaultLoadBalancedWebService()
case RequestDrivenWebServiceType:
	m := newDefaultRequestDrivenWebService()
// ... other types.
default:
	return nil, &ErrInvalidWorkloadType{Type: typeVal}
}
if err := yaml.Unmarshal(in, m); err != nil {
	return nil, fmt.Errorf("unmarshal manifest for %s: %w", typeVal, err)
}
if err := m.windowsCompatibility(); err != nil {
	return nil, err
}
return m, nil
```

Reply:
Now that `windowsCompatibility()` isn't in the interface, RDWSType no longer has to call it to satisfy the interface, so only 4 of the 5 workload types is calling it...and I can't extract `Unmarshal` because that has to precede `windowsCompatibility()` ☹️

Reply:
Reinstated the method for RDWS and did the refactor. 👍🏼